### DISCUSSION TITLE
Use with ethers.js

### DISCUSSION THREAD

**Discussion by Digtyarenk0** [alchemyplatform/aa-sdk#824] [Issues] [answered]
Hello, is it possible to work with ethers through an abstract account?
Is it possible to somehow connect the resulting ModularAccountAlchemyClient from createModularAccountAlchemyClient to a contract?
I would like to connect ModularAccountAlchemyClient to a contract on the frontend in Contract from ethers/lib/ethers 
For example

> ```
>  const contract = new Contract(
>  "0xeBDCbab722f9B4614b7ec1C261c9E52acF109CF8",
>  ERC20__factory.abi,
>  ModularAccountAlchemyClient.provider
>  );
> 
> ```

To call contract methods on behalf of an abstract account. And further calling contract methods with new Contract from ethers. For example like this:
> ```
>     const contract = new Contract(
>         "0xeBDCbab722f9B4614b7ec1C261c9E52acF109CF8",
>         ERC20__factory.abi,
>         ModularAccountAlchemyClient.provider
>         );
>         const tx = await contract.functions['approve']('0xeBDCbab722f9B4614b7ec1C261c9E52acF109CF8','10000');
> ```

**Comment by avarobinson** [comment, marked as answer]
Hi @Digtyarenk0! Have you taken a look at our [aa-ethers](https://accountkit.alchemy.com/packages/aa-ethers/) package?
